### PR TITLE
Adding ability to run pre-compiled .NET assemblies

### LINKED ISSUE #943: Support pre-compiled functions
Enable the ability to define a function that uses a pre-compiled .NET assembly.

For the initial implementation, filename (assembly file name) and entry points will be defined in the existing function metadata properties `scriptFile` and `entryPoint`

### PR DESCRIPTION
**Pull request by fabiocav**
Resolves #943 

This enables running functions from pre-compiled .NET assemblies.

This initial implementation leverages existing function metadata properties to define the assembly file name (we still, by convention, respect a file named `run.dll`) and entry point:

```
...
"scriptFile":"DotNetFunctionAssembly.dll",
"entryPoint": "TestFunction.Function.Run"
...
```
Will be adding additional tests to this.

### CODE REVIEW

**Comment by mathewc** [comment]
:ship:

File: test/WebJobs.Script.Tests.Integration/DotNetEndToEndTests.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+                string primaryReferenceSource = @"
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace TestFunction
+{
+    public class Function
```
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/DotNetEndToEndTests.cs]: might as well colocate the source and the json - either both in source, or both in resx

**Comment by dsyme** [comment]
Nice job!

**Comment by mikezks** [comment]
nice feature, thanks!

this needs to be added to the function.json schema.

**Pull request merged**